lib/matplotlib/backends/registry.py

```diff
@@ -0,0 +1,61 @@
+from enum import Enum
+
+
+class BackendFilter(Enum):
+    INTERACTIVE = 0
+    INTERACTIVE_NON_WEB = 1
+    NON_INTERACTIVE = 2
+
+
+class BackendRegistry:
+    """
+    Registry of backends available within Matplotlib.
+
+    This is the single source of truth for available backends.
+    """
+    def __init__(self):
+        # Built-in backends are those which are included in the Matplotlib repo.
+        # A backend with name 'name' is located in the module
+        # f'matplotlib.backends.backend_{name.lower()}'
+
+        # The capitalized forms are needed for ipython at present; this may
+        # change for later versions.
+        self._builtin_interactive = [
+            "GTK3Agg", "GTK3Cairo", "GTK4Agg", "GTK4Cairo",
+            "MacOSX",
+            "nbAgg",
+            "QtAgg", "QtCairo", "Qt5Agg", "Qt5Cairo",
+            "TkAgg", "TkCairo",
+            "WebAgg",
+            "WX", "WXAgg", "WXCairo",
+        ]
+        self._builtin_not_interactive = [
+            "agg", "cairo", "pdf", "pgf", "ps", "svg", "template",
+        ]
+        self._framework_to_backend_mapping = {
+            "qt": "qtagg",
+            "gtk3": "gtk3agg",
+            "gtk4": "gtk4agg",
+            "wx": "wxagg",
+            "tk": "tkagg",
+            "macosx": "macosx",
+            "headless": "agg",
+        }
+
+    def framework_to_backend(self, interactive_framework):
```

Comment:
We can afford quite a long name for more clarity. 
```suggestion
    def gui_framework_to_backend(self, interactive_framework):
```
Or possibly even better
```suggestion
    def backend_for_gui_framework(self, interactive_framework):
```
You may then also shorten the parameter to just `framework` because *interactive* does not any information.

Reply:
I prefer the last suggestion.